src/server/hotkey_collector.cpp

```diff
@@ -28,10 +36,18 @@ void hotkey_collector::handle_rpc(const dsn::replication::detect_hotkey_request
 
 void hotkey_collector::capture_raw_key(const dsn::blob &raw_key, int64_t weight)
 {
-    // TODO: (Tangyanzhao) Add a judgment sentence to check if it is a raw key
+    dsn::blob hash_key, sort_key;
+    pegasus_restore_key(raw_key, hash_key, sort_key);
+    capture_hash_key(hash_key, weight);
+}
+
+void hotkey_collector::capture_hash_key(const dsn::blob &hash_key, int64_t weight)
+{
+    // TODO: (Tangyanzhao) add a unit test to ensure data integrity
```

Comment:
What is the data integrity to be tested? and how about add the unit test in the pull request?

Reply:
Ok, I will add a function test to test it in the future PR